packages/workflow/src/features/registration/handler.ts

```diff
@@ -239,182 +218,65 @@ export async function createRegistrationHandler(
     request.payload as IEventRegistrationCallbackPayload
 
   if (error) {
     throw new Error(`Callback triggered with an error: ${error}`)
   }
 
-  const taskBundle: Bundle = await getFromFhir(`/Task?identifier=${trackingId}`)
-  if (
-    !taskBundle ||
-    !taskBundle.entry ||
-    !taskBundle.entry[0] ||
-    !taskBundle.entry[0].resource
-  ) {
-    throw new Error(
-      `Task for tracking id ${trackingId} could not be found during markEventAsRegisteredCallbackHandler`
-    )
-  }
-
-  const task = taskBundle.entry[0].resource as Task
-  if (!task.focus || !task.focus.reference) {
-    throw new Error(`Task ${task.id} doesn't have a focus reference`)
-  }
-  const composition: Composition = await getFromFhir(`/${task.focus.reference}`)
-  const event = getTaskEventType(task) as EVENT_TYPE
-
   try {
-    await markEventAsRegistered(
-      task,
-      registrationNumber as RegistrationNumber,
-      event,
+    const savedRecord = await getRecordById(
+      compositionId,
+      request.headers.authorization,
+      ['WAITING_VALIDATION']
+    )
+    if (!savedRecord) {
+      throw new Error('Could not find record in elastic search!')
+    }
+    const practitioner = await getLoggedInPractitionerResource(
       getToken(request)
     )
 
-    /** pushing registrationNumber on related person's identifier
-     *  taking patients as an array because MARRIAGE Event has two types of patient
-     */
-    const patients: Patient[] = await updatePatientIdentifierWithRN(
-      composition,
-      SECTION_CODE[event],
-      REG_NUMBER_SYSTEM[event],
-      registrationNumber
-    )
+    const event = getEventType(savedRecord)
+    const eventNotification = isEventNotification(savedRecord)
 
-    if (event === EVENT_TYPE.BIRTH && childIdentifiers) {
-      // For birth event patients[0] is child and it should
-      // already be initialized with the RN identifier
-      childIdentifiers.forEach((childIdentifier) => {
-        const previousIdentifier = patients[0].identifier!.find(
-          ({ type }) => type?.coding?.[0].code === childIdentifier.type
-        )
-        if (!previousIdentifier) {
-          patients[0].identifier!.push({
-            type: {
-              coding: [{ code: childIdentifier.type }]
-            },
-            value: childIdentifier.value
-          })
-        } else {
-          previousIdentifier.value = childIdentifier.value
-        }
-      })
-    }
+    const registeredBundle = await toRegistered(
+      request,
+      savedRecord,
+      practitioner,
+      registrationNumber,
+      childIdentifiers
+    )
 
-    if (event === EVENT_TYPE.DEATH) {
-      /** using first patient because for death event there is only one patient */
-      patients[0] = await validateDeceasedDetails(patients[0], {
-        Authorization: request.headers.authorization
-      })
+    if (!registeredBundle) {
+      throw new Error('Could not create registered bundle!')
```

Comment:
Can registeredBundle be null in some case?

Reply:
actually not, this is redundant